src/MobileBookingHeader/MobileBookingHeader.js

```diff
@@ -0,0 +1,120 @@
+// @flow
+
+import * as React from 'react';
+import { ChevronDown, ChevronUp } from '@kiwicom/orbit-components/lib/icons';
+import css from 'styled-jsx/css';
+
+import responsiveStyleHelperClasses from '../common/responsiveStyleHelperClasses';
+import { BookingState } from '../context/BookingState';
+import MobileNearestBooking from './MobileNearestBooking';
+import MobileSelectedBooking from './MobileSelectedBooking';
+
+type MobileBookingPageProps = {|
+  bookingPage: string,
+  selectedBooking: ?string,
+  expanded: boolean,
+|};
+
+const MobileBookingPage = (props: MobileBookingPageProps) => {
+  const { bookingPage, selectedBooking, expanded } = props;
+  if (bookingPage === 'SINGLE_BOOKING') {
+    if (selectedBooking) {
+      return (
+        <MobileSelectedBooking
+          bookingId={selectedBooking}
+          expanded={expanded}
+        />
+      );
+    }
+
+    return <MobileNearestBooking expanded={expanded} />;
+  }
+
+  return null;
+};
+
+type MobileBookingSummaryProps = {
+  style: any,
+  expanded: boolean,
+};
+const MobileBookingSummary = (props: MobileBookingSummaryProps) => (
+  <div style={props.style}>
+    <BookingState.Consumer>
+      {({ bookingPage, selectedBooking }) => (
+        <MobileBookingPage
+          bookingPage={bookingPage}
+          selectedBooking={selectedBooking}
+          expanded={props.expanded}
+        />
+      )}
+    </BookingState.Consumer>
+  </div>
+);
+
+const MobileBookingHeaderStyle = css`
+  .MobileBookingHeader {
+    display: flex;
+    flex-direction: column;
+    background-color: #f5f7f9;
+    box-shadow: inset 0 1px 0 0 #e8edf1;
+    padding: 0 16px;
+  }
+
+  .topRow {
+    display: flex;
+  }
+
+  .Chevron {
+    display: flex;
+    flex-direction: column;
+    justify-content: space-around;
+    cursor: pointer;
+  }
+`;
+
+type Props = {};
+type State = {
+  expanded: boolean,
+};
+
+class MobileBookingHeader extends React.Component<Props, State> {
+  state = {
+    expanded: false,
+  };
+
+  toggle() {
+    this.setState({ expanded: !this.state.expanded });
```

Comment:
I know it works like this, but what I think @mrtnzlml was trying to say the other day is that this implementation is better for this cases, `this.setState(prevState => ({ expanded: !prevState.expanded }))`

Reply:
Yes, this can be safely considered to be a bug: https://reactjs.org/docs/react-component.html#setstate

You should set an eslint rule to prevent this... :)